app/modules/site_settings/resources.py

```diff
@@ -0,0 +1,133 @@
+class SiteSettings(Resource):
+    """
+    Manipulations with Site Settings.
+    """
+
+    @api.permission_required(permissions.AdminRolePermission())
+    @api.parameters(PaginationParameters())
+    @api.response(schemas.BaseSiteSettingSchema(many=True))
+    def get(self, args):
+        """
+        List of SiteSetting.
+
+        Returns a list of SiteSetting starting from ``offset`` limited by ``limit``
+        parameter.
+        """
+        return (
+            SiteSetting.query.order_by('key').offset(args['offset']).limit(args['limit'])
+        )
+
+    @api.permission_required(permissions.AdminRolePermission())
+    @api.login_required(oauth_scopes=['site-settings:write'])
+    @api.parameters(parameters.CreateSiteSettingParameters())
+    @api.response(schemas.DetailedSiteSettingSchema())
+    @api.response(code=HTTPStatus.CONFLICT)
+    def post(self, args):
+        """
+        Create or update a SiteSetting.
+        """
+        if args.get('transactionId'):
+            transaction_id = args.pop('transactionId')
+            if args.get('transactionPath'):
+                paths = [args.pop('transactionPath')]
+            else:
+                paths = None
+            fups = (
+                FileUpload.create_fileuploads_from_tus(transaction_id, paths=paths) or []
+            )
+            if len(fups) != 1:
+                # Delete the files in the filesystem
+                # Can't use .delete() because fups are not persisted
+                for fup in fups:
+                    path = Path(fup.get_absolute_path())
+                    if path.exists():
+                        path.unlink()
+
+                abort(
+                    code=HTTPStatus.UNPROCESSABLE_ENTITY,
+                    message=f'Transaction {transaction_id} has {len(fups)} files, need exactly 1.',
+                )
+            with db.session.begin():
+                db.session.add(fups[0])
+            args['file_upload_guid'] = fups[0].guid
+        site_setting = SiteSetting.set(**args)
+        return site_setting
+
+
+@api.route('/<string:site_setting_key>')
+@api.response(
+    code=HTTPStatus.NOT_FOUND,
+    description='SiteSetting not found.',
+)
+@api.resolve_object_by_model(SiteSetting, 'site_setting', 'site_setting_key')
+class SiteSettingByKey(Resource):
+    """
+    Manipulations with a specific SiteSetting.
+    """
+
+    @api.permission_required(
+        permissions.ObjectAccessPermission,
+        kwargs_on_request=lambda kwargs: {
+            'obj': kwargs['site_setting'],
+            'action': AccessOperation.READ,
+        },
+    )
+    def get(self, site_setting):
+        """
+        Get SiteSetting details by ID.
+        """
+        return redirect(
+            url_for(
+                'api.fileuploads_file_upload_src_u_by_id_2',
+                fileupload_guid=site_setting.file_upload_guid,
+            )
+        )
+
+    @api.permission_required(permissions.AdminRolePermission())
```

Comment:
Should use ObjectAccessPermission for object permissions

Reply:
I wanted only admins to edit, create and delete site settings.  Is that supposed to be set somewhere else?  I have looked at ObjectAccessPermission and I don't really know how the permissions stuff work.

Reply:
"Admins" is a vague concept in Houston and worth avoiding. Have you seen the roles spreadsheet? 
That should give you some idea of why it's confusing :)
Happy to go through it with you UK time tomorrow.